### ISSUE TITLE
[Feature Request]: Improved keyboard navigation

### ISSUE THREAD

**Issue by Cruxial0** [Particle1904/DatasetHelpers#24]
Been using this program for editing my datasets recently. It's great, however I feel it lacks in terms of keyboard navigation. With tag editing being a very keyboard intensive task, it makes sense to allow the user to perform some actions using the keyboard.

Suggested keyboard shortcuts (Tag/Caption Editor) (Windows/Linux):
- ~Move 1 image forward: ALT + Arrow Right~
- ~Move 1 image back: ALT + Arrow Left~
- Adjust tag weight: CTRL + Arrow Up/Down (similar to a1111 - in increments of 0.1)

EDIT: ALT + Arrow Keys is already implemented, oops. As a quality of life improvement to this bind, the caret should be set to either the end of the text, or in a position relative to where you were on the previous page.

EDIT 2: Sometimes when using the ALT + Arrow Keys keybind, the letter 'B' is added to the start of the prompt. It seemingly only happens after editing text THEN swapping pages.

**Comment by Particle1904** [comment]
I noticed that and I think its a bug in the keyboard library I'm using since it didn't happen when I first implemented the feature (but I updated the dependency to a new version).

**Comment by Particle1904** [comment]
I've changed it so the caret position stays in the same position after changing images (in terms of characters count from the beginning of the text) or snaps to the end if the caret position is bigger than the last image text size. It will also focus on the Editor control to avoid users having to manually click on it to continue editing text.

~Seems like the random "B" being added is still occurring in the current dev commit and I have no idea why. Its extremely hard to debug that since its so random. I think its something to do with Alt + Codes... I've noticed it also adding an "X", a "," and even a "▬"~. I tried:
 - Disabling the Editor Control when the Navigate command is called.
 - Focusing on a different control before the Navigation command.
 - Manually handling the KeyDown and KeyUp events in Avalonia.
 - And others.
**I tried 11 different possible solutions and all of them failed.**
 
It seems like this is a default feature of Avalonia, I found only a single issue in their GitHub about this and there's a single response from another user. Basically if you type any text into a text Control then do any type of Alt Code combination, it adds the "special" character, for example typing "a" then pressing Alt + 2 + 5 + 5 adds a "7" in the text box... as a workaround, F1 to F6 keys (and Mouse with Key 4 and Key 5) can be used to navigate between images as described [here](https://github.com/Particle1904/DatasetHelpers/wiki/Editor-Page#functionality).

I have no plans to support editing of weighted captions/tags, so Alt + Up and Alt + Down won't be added. I added generating tags with weighs a while back and never tried training a model with it, and since its something that's not that popular/common, I wont add support.
